lib/SIL/SILValue.cpp

```diff
@@ -58,22 +58,25 @@ void ValueBase::replaceAllUsesWithUndef() {
 }
 
 SILInstruction *ValueBase::getDefiningInstruction() {
-  if (auto inst = dyn_cast<SingleValueInstruction>(this))
+  if (auto *inst = dyn_cast<SingleValueInstruction>(this))
     return inst;
-  // TODO: MultiValueInstruction
+  if (auto *result = dyn_cast<MultipleValueInstructionResult>(this))
+    return result->getParent();
   return nullptr;
 }
 
 Optional<ValueBase::DefiningInstructionResult>
 ValueBase::getDefiningInstructionResult() {
-  if (auto inst = dyn_cast<SingleValueInstruction>(this))
+  if (auto *inst = dyn_cast<SingleValueInstruction>(this))
     return DefiningInstructionResult{ inst, 0 };
-  // TODO: MultiValueInstruction
+  if (auto *result = dyn_cast<MultipleValueInstructionResult>(this))
+    return DefiningInstructionResult{result->getParent(), result->getIndex()};
```

Comment:
The style immediately above is to put spaces after the ``{``; you should be consistent one way or the other.